backend-rust/src/indexer.rs

```diff
@@ -876,7 +915,182 @@ WHERE blocks.finalization_time IS NULL AND blocks.height <= last.height
+                    transaction_type: AccountStatementEntryType::FoundationReward,
+                },
+                PreparedBlockSpecialEvent {
+                    account_address: baker.to_string(),
+                    amount: baker_reward.micro_ccd.try_into()?,
+                    block_height: block_height.height.try_into()?,
+                    slot_time,
+                    transaction_type: AccountStatementEntryType::BakerReward,
+                },
+            ],
+            SpecialTransactionOutcome::PaydayFoundationReward {
+                foundation_account,
+                development_charge,
+            } => vec![PreparedBlockSpecialEvent {
+                account_address: foundation_account.to_string(),
+                amount: development_charge.micro_ccd.try_into()?,
+                block_height: block_height.height.try_into()?,
+                slot_time,
+                transaction_type: AccountStatementEntryType::FoundationReward,
+            }],
+            SpecialTransactionOutcome::PaydayAccountReward {
+                account,
+                transaction_fees,
+                baker_reward,
+                finalization_reward,
+            } => vec![
+                PreparedBlockSpecialEvent {
+                    account_address: account.to_string(),
+                    amount: transaction_fees.micro_ccd.try_into()?,
+                    block_height: block_height.height.try_into()?,
+                    slot_time,
+                    transaction_type: AccountStatementEntryType::TransactionFeeReward,
+                },
+                PreparedBlockSpecialEvent {
+                    account_address: account.to_string(),
+                    amount: baker_reward.micro_ccd.try_into()?,
+                    block_height: block_height.height.try_into()?,
+                    slot_time,
+                    transaction_type: AccountStatementEntryType::BakerReward,
+                },
+                PreparedBlockSpecialEvent {
+                    account_address: account.to_string(),
+                    amount: finalization_reward.micro_ccd.try_into()?,
+                    block_height: block_height.height.try_into()?,
+                    slot_time,
+                    transaction_type: AccountStatementEntryType::FinalizationReward,
+                },
+            ],
+            SpecialTransactionOutcome::BlockAccrueReward {
+                ..
+            }
+            | SpecialTransactionOutcome::PaydayPoolReward {
+                ..
+            } => Vec::new(),
+        };
+        Ok(results)
+    }
+
+    async fn save(
+        &self,
+        tx: &mut sqlx::Transaction<'static, sqlx::Postgres>,
+    ) -> anyhow::Result<()> {
+        sqlx::query!(
+            "WITH account_info AS (
+            SELECT index AS account_index, amount + $4 AS current_balance
+            FROM accounts
+            WHERE address = $1
+        )
+        INSERT INTO account_statements (
```

Comment:
We should also probably track transaction fees and the amount part of different types of transfers in this table, but maybe that is for another PR, please add a task in that case.

Reply:
I will add task for transaction fees, 

For the other types of transfers I will add those as part of this PR.

Reply:
https://linear.app/concordium/issue/CCD-138/accountstatementtransactionfees